volatility/framework/plugins/windows/hashdump.py

```diff
@@ -0,0 +1,346 @@
+        d1 = DES.new(des_k1.encode('latin1'), DES.MODE_ECB)
+        d2 = DES.new(des_k2.encode('latin1'), DES.MODE_ECB)
+        md5 = MD5.new()
+
+        md5.update(hbootkey[:0x10] + pack("<L", rid) + lmntstr)
+        rc4_key = md5.digest()
+        rc4 = ARC4.new(rc4_key)
+        obfkey = rc4.encrypt(enc_hash)
+
+        hash = d1.decrypt(obfkey[:8]) + d2.decrypt(obfkey[8:])
+        return hash
+
+
+
+    def get_user_name(self, user, samhive):
+        V = None
+        for v in user.get_values():
+            if v.get_name() == 'V':
+                V = samhive.read(v.Data+4, v.DataLength)
+        if not V:
+            return None
+
+        name_offset = unpack("<L", V[0x0c:0x10])[0] + 0xCC
+        name_length = unpack("<L", V[0x10:0x14])[0]
+        if name_length > len(V):
+            return None
+
+        username = V[name_offset:name_offset + name_length].decode('utf-16-le')
+        return username
+
+    #replaces the dump_hashes method in vol2
+    def _generator(self, syshive, samhive):
+        if syshive == None:
+            print("SYSTEM address is None: Did you use the correct profile?")
+            yield self.none_obj
+        if samhive == None:
+            print("SAM address is None: Did you use the correct profile?")
+            yield self.none_obj
+        bootkey = self.get_bootkey(syshive)
+        hbootkey = self.get_hbootkey(samhive, bootkey)
+        if hbootkey:
+            for user in self.get_user_keys(samhive):
+                ret = self.get_user_hashes(user, samhive, hbootkey)
+                if not ret:
+                    yield self.none_obj
+                else:
+                    lmhash, nthash = ret
+                    if not lmhash:
+                        lmhash = self.empty_lm
+                    if not nthash:
+                        nthash = self.empty_nt
+                        ## temporary fix to prevent UnicodeDecodeError backtraces 
+                        ## however this can cause truncated user names as a result
+                    name = self.get_user_name(user, samhive)
+                    if name is not None:
+                        name = name.encode('ascii', 'ignore')
+                    else:
+                        name = "(unavailable)"
+                        
+
+                    if lmhash == self.empty_lm:
+                        lmout=lmhash
+                    else:
+                        lmout = binascii.hexlify(lmhash).decode('latin1')
+                        
+                    if nthash == self.empty_nt:
+                        ntout=nthash
+                    else:
+                        ntout = binascii.hexlify(nthash).decode('latin1')
+                    yield (0,   (name.decode('latin1'), 
+                                str(int(str(user.get_name()), 16)),
+                                lmout,
+                                ntout))
+        else:
+            print("Hbootkey is not valid")
+            yield self.none_obj
```

Comment:
This will end the generator, it seems better to throw an (informative) exception or error here.

Also, please use the logging framework rather than directly printing anything.  This is designed to be a library and the console output may never get seen if it's wired into a webpage or similar...